### ISSUE TITLE
IDE1006 shouldn't report on top-level `Main` method with `await` operators

### ISSUE THREAD

**Issue by SunnieShine** [dotnet/roslyn#49648]
**Version Used**: 
VS 16.8.2

**Steps to Reproduce**:

1. Write a program using top-level statements, such as:
```csharp
// Program.cs
using System;

var sessionSettings = new SessionSettings("127.0.0.1", 8080, "1234567890");
await using var session = new Session(sessionSettings, myNumber);
await session.ConnectAsync();

Console.WriteLine("Successful!");
```

2. IDE1006 (naming styles) suggests.

**Expected Behavior**:
Because of implicit `Main` statement, IDE1006 shouldn't report on this scenario. In fact, we can't modify the suffix of the name of the `Main` method explicitly.

**Actual Behavior**:
IDE1006 raised, and all codes are marked with green waves.

**Comment by Youssef1313** [comment]
Probably the analyzer should call `CSharpSyntaxFacts.IsValidIdentifier` method:

https://github.com/dotnet/roslyn/blob/782c293d849c060d435b66488206f2e78dd92e4e/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs#L428

instead of the following check:

https://github.com/dotnet/roslyn/blob/a6144b2b6accf6c9b0fce4769440b6f0f1bcbb1b/src/Analyzers/Core/Analyzers/NamingStyle/NamingStyleDiagnosticAnalyzerBase.cs#L110-L113

```diff
-             if (string.IsNullOrEmpty(symbol.Name))
+             if (!CSharpSyntaxFacts.IsValidIdentifier(symbol.Name))
```

**Comment by huoyaoyuan** [comment]
Consider #49566 :
Does generated main have CanBeReferencedByName = false? If so, this case should already be covered.

You may add a new test in master branch to verify this.

**Comment by huoyaoyuan** [comment]
@SunnieShine can you confirm if this still repro in 16.9?

**Comment by SunnieShine** [comment]
> @SunnieShine can you confirm if this still repro in 16.9?

@huoyaoyuan Hmmm actually 16.9 and 16.10 doesn't warn about this, but this issue is still open... I don't know whether the bug has already been fixed.

**Comment by huoyaoyuan** [comment]
If you can confirm it doesn't warn any more, I think it should be fixed by #49566, and the issue can be closed.

**Comment by SunnieShine** [comment]
@huoyaoyuan Thanks for your mention.

**Comment by Youssef1313** [comment]
I'm adding tests in #51831 to prevent regressions.
